Rubberduck.InternalApi/Common/FileSystemProvider.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.IO.Abstractions;
+
+namespace Rubberduck.InternalApi.Common
+{
+    public static class FileSystemProvider
+    {
+        static FileSystemProvider()
+        {
+            FileSystem = new FileSystem();
+        }
+
+        public static IFileSystem FileSystem { get; internal set; }
```

Comment:
I should add a comment. This is meant to support unit tests on the objects that are not part of the IoC container by allowing the mocking subsystem to inject the mocked `IFileSystem` into the provider that are used by such objects. Otherwise, it has no use in the production.

Reply:
Later, I saw that we use it in the `VbComponent`. One thought for a future PR might be to add a factory for those and inject the filesystem into the factory.

Reply:
How would that be better/different from the provider with a static field? Considering that in production, there really can be only one instance of the `IFileSystem`, which is the actual filesystem, a static field seemed a more accurate description of the reality?

Reply:
For now, a comment is in place. Hopefully that will be adequate unless I can get an example of how a factory would be better than a provider here.